### ISSUE TITLE
TF Apply with 2.6.0 provider results in "Error: Provider produced inconsistent final plan"

### ISSUE THREAD

**Issue by gcardonag** [hashicorp/terraform-provider-helm#892]
### Terraform, Provider, Kubernetes and Helm Versions

```
Terraform version: v1.1.6
Provider version: v2.6.0
Kubernetes version: v1.19
```

### Affected Resource(s)

- helm_release

### Terraform Configuration Files
```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

```
resource "helm_release" "gitlab" {
  name       = "gitlab"
  repository = "https://charts.gitlab.io/"
  chart      = "gitlab"
  # To see how the Helm chart version maps to a GitLab version, see:
  # https://docs.gitlab.com/charts/installation/version_mappings.html
  version    = "5.9.4"
  namespace  = kubernetes_namespace.gitlab.metadata[0].name
  wait = false # Waiting seems to stall regardless of whether all completion conditions are met successfully

  values = [ yamlencode(local.gitlab_chart_values) ]
}
```

### Debug Output

NOTE: In addition to Terraform debugging, please set HELM_DEBUG=1 to enable debugging info from helm.

### Steps to Reproduce

1. `terraform apply` with a state that existed pre-2.6.0 provider release.

### Expected Behavior

I expected `terraform apply` to complete without any errors.

### Actual Behavior

After `terraform apply` completes, I see the following error:
```
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for helm_release.gitlab to include new values
│ learned so far during apply, provider
│ "registry.terraform.io/hashicorp/helm" produced an invalid new value for
│ .pass_credentials: was cty.False, but now null.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
```

### References

- https://github.com/hashicorp/terraform-provider-helm/releases/tag/v2.6.0
- https://github.com/hashicorp/terraform-provider-helm/pull/841

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jrhouston** [comment]
Thanks for opening this @gcardonag. I'm having some trouble reproducing this. These are the steps I'm using:

1. `terraform apply` a chart using v2.5.1 of the provider
2. Switch to v2.6.0 and run `terraform init -upgrade`
3. Make a change to the config and run `terraform apply` again.

I see the line in the diff adding the new field:

```
+ pass_credentials           = false
```

But I don't get any error, and the apply runs successfully. I tried pinning to the terraform version you're using in case it was a terraform version issue but no luck. Even if I manually edit the state file and set `pass_credentials` to `null` I don't get the error when I run apply. 

Would you be able to run `TF_LOG=DEBUG terraform apply` and share the output to see if that gives us any clues.

**Comment by gcardonag** [comment]
@jrhouston - I tried a run with `TF_LOG=DEBUG`. Here are the segments I found that may be helpful (I trimmed for brevity and to protect sensitive information that is hard to parse through):
```
2022-06-19T20:02:24.453Z [DEBUG] After incorporating new values learned so far during apply, helm_release.gitlab change has become NoOp
2022-06-19T20:02:24.454Z [ERROR] vertex "helm_release.gitlab" error: Provider produced inconsistent final plan
2022-06-19T20:02:24.465Z [DEBUG] Uploading remote state to S3: {
...
PUT ....tfstate HTTP/1.1
{
      "mode": "managed",
      "type": "helm_release",
      "name": "gitlab",
      "provider": "provider[\"registry.terraform.io/hashicorp/helm\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "atomic": false,
            "chart": "gitlab",
            "cleanup_on_fail": false,
            "create_namespace": false,
            "dependency_update": false,
            "description": null,
            "devel": null,
            "disable_crd_hooks": false,
            "disable_openapi_validation": false,
            "disable_webhooks": false,
            "force_update": false,
            "id": "gitlab",
            "keyring": null,
            "lint": false,
            "manifest": null,
            "max_history": 0,
            "metadata": [
              {
                "app_version": "...",
                "chart": "gitlab",
                "name": "gitlab",
                "namespace": "gitlab",
                "revision": 47,
                "values": "...",
                "version": "5.9.4"
              }
            ],
            "name": "gitlab",
            "namespace": "gitlab",
            "pass_credentials": null,
            "postrender": [],
            "recreate_pods": false,
            "render_subchart_notes": true,
            "replace": false,
            "repository": "https://charts.gitlab.io/",
            "repository_ca_file": null,
            "repository_cert_file": null,
            "repository_key_file": null,
            "repository_password": null,
            "repository_username": null,
            "reset_values": false,
            "reuse_values": false,
            "set": [],
            "set_sensitive": [],
            "skip_crds": false,
            "status": "deployed",
            "timeout": 300,
            "values": [...],
            "verify": false,
            "version": "...",
            "wait": false,
            "wait_for_jobs": false
          },
          "sensitive_attributes": [...],
          "private": "...",
          "dependencies": [...]
        }
      ]
    },
...
2022-06-19T20:02:14.982Z [WARN]  Provider "registry.terraform.io/hashicorp/helm" produced an invalid plan for helm_release.gitlab, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .disable_crd_hooks: planned value cty.False for a non-computed attribute
      - .disable_openapi_validation: planned value cty.False for a non-computed attribute
      - .create_namespace: planned value cty.False for a non-computed attribute
      - .pass_credentials: planned value cty.False for a non-computed attribute
      - .replace: planned value cty.False for a non-computed attribute
      - .disable_webhooks: planned value cty.False for a non-computed attribute
      - .recreate_pods: planned value cty.False for a non-computed attribute
      - .dependency_update: planned value cty.False for a non-computed attribute
      - .reset_values: planned value cty.False for a non-computed attribute
      - .skip_crds: planned value cty.False for a non-computed attribute
      - .verify: planned value cty.False for a non-computed attribute
      - .cleanup_on_fail: planned value cty.False for a non-computed attribute
      - .render_subchart_notes: planned value cty.True for a non-computed attribute
      - .lint: planned value cty.False for a non-computed attribute
      - .timeout: planned value cty.NumberIntVal(300) for a non-computed attribute
      - .force_update: planned value cty.False for a non-computed attribute
      - .values: planned value cty.UnknownVal(cty.List(cty.String)) does not match config value...
      - .wait_for_jobs: planned value cty.False for a non-computed attribute
      - .atomic: planned value cty.False for a non-computed attribute
      - .reuse_values: planned value cty.False for a non-computed attribute
      - .max_history: planned value cty.NumberIntVal(0) for a non-computed attribute
```

**Comment by com30n** [comment]
I have the same issue:
```
│Error: Provider produced inconsistent final plan
│
│When expanding the plan for
│module.helm_addon.helm_release.addon[0] to
│include new values learned so far during apply, provider
│"registry.terraform.io/hashicorp/helm" produced an invalid new value for
│.pass_credentials: was cty.False, but now null.
│
│This is a bug in the provider, which should be reported in the provider's
│own issue tracker.
```

**Comment by WarpRat** [comment]
@gcardonag - do you have the experimental manifest feature on? We were running into this issue with that enabled (on a different chart, can't recall which one) and turning it off allowed the chart to plan and apply as normal. It's a shame because we really miss the much better diff we get from that feature but between this issue and problems in #771 it was generating too many support requests to our SRE team to leave on.

**Comment by mymasse** [comment]
@WarpRat we don't have that feature enabled and run into this issue

**Comment by severity1** [comment]
I am experiencing the same issue

**Comment by njgibbon** [comment]
Hello,

Today I've experienced the same issue whilst trying to go from `v2.5.1` to `v2.6.0`.

**Comment by sajithvasu** [comment]
I'm experiencing the same issue with v2.6.0 and I followed the work-around mentioned in this link - https://github.com/hashicorp/terraform-provider-helm/issues/752 to update state file manually for pass_credentials and set it to false.

**Comment by muram** [comment]
Manually setting `pass_credentials` to `false` in tfstate file after the upgrade then `terraform apply` worked for me as a workaround.

**Comment by jdpribula** [comment]
I had a similar issue with **_set_** reverting to 2.5.1 resolved the issue for me.

**Comment by kbyri-stats** [comment]
A similar issue has been experienced by me many times while performing terraform apply. downgrading helm version to 2.5.1 solved the issue. But IMHO this is just a work around which is not the actual fix. I am expecting a perfect resolution from contributors soon.

**Comment by 9numbernine9** [comment]
This still appears to be an issue in `2.7.0`. Running into the same error that @com30n [reported above](https://github.com/hashicorp/terraform-provider-helm/issues/892#issuecomment-1167129383).

**Comment by iamdevnull** [comment]
Issue is still present and came probably with 

hashicorp/helm - v2.6.0 - https://github.com/hashicorp/terraform-provider-helm/releases 

`Add pass_credentials attribute (#841)`

This is my workaround:

```
terraform state pull > /tmp/tfm-state.json

sed -i 's#"pass_credentials": null#"pass_credentials": false#g' /tmp/tfm-state.json

current_serial=$(grep -w 'serial' /tmp/tfm-state.json | cut -d ":" -f 2 | tr -d ',' | tr -d "[:space:]") && new_serial=$(expr ${current_serial} + 1) && sed -i "s#\"serial\": ${current_serial}#\"serial\": ${new_serial}#g" /tmp/tfm-state.json

terraform state push /tmp/tfm-state.json
```

Maybe this helps someone.

**Comment by msolimans** [comment]
had pretty similar issue but it was more relative to `manifest` - unfortunately, I had to turn off experimental manifest generation  (hope this can help others in case anyone stumbles upon same issue) 
```
  experiments {
    manifest = true
  }
```
